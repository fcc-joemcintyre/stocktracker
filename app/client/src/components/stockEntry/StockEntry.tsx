import { useState } from 'react';

type Props = {
  onAddStock: (symbol: string) => void,
};

export const StockEntry = ({
  onAddStock,
}: Props) => {
  const [symbol, setSymbol] = useState ('');

  function onChange (e) {
    setSymbol (e.target.value);
  }

  function onSubmit (e) {
    e.preventDefault ();
    const t = symbol.trim ().toUpperCase ();
    onAddStock (t);
    setSymbol ('');
  }

  return (
    <form onSubmit={onSubmit}>
      <label htmlFor='symbolText'>Symbol:</label>
      {' '}
      <input
        type='text'
        id='symbolText'
        value={symbol}
        style={{ width: '60px' }}
        onChange={onChange}
      />
      {' '}
      <button
        type='submit'
        disabled={(symbol.trim () === '')}
      >
        Track
      </button>
    </form>
  );
};
