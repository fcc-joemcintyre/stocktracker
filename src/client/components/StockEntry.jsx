/**
 * Copyright (c) Joe McIntyre, 2016
 * license: MIT (https://github.com/fcc-joemcintyre/stocktracker/LICENSE.txt)
 */
import React from "react";

export default class StockEntry extends React.Component {
  constructor (props) {
    super (props);
    this.state = {
      symbol: ""
    }
    this.onChange = this.onChange.bind (this);
    this.onSubmit = this.onSubmit.bind (this);
  }

  onChange (event) {
    this.setState ({symbol:event.target.value});
  }

  onSubmit (event) {
    event.preventDefault ();
    let symbol = this.state.symbol.trim ().toUpperCase ();
    this.props.addStock (symbol);
    this.setState ({symbol: ""});
  }

  render() {
    return (
      <form className="entryBox" onSubmit={this.onSubmit}>
        <label id="symbolText">Symbol:</label>
        <input type="text" id="symbolText" className="entryText" value={this.state.symbol} onChange={this.onChange}/>
        <input type="submit" className="entryButton"
          disabled={(this.state.symbol === "")}
          value="Track"/>
      </form>
    );
  }
}
