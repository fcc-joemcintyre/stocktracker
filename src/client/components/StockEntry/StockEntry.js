import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const StockEntry = ({ onAddStock }) => {
  const [symbol, setSymbol] = useState ('');

  function onChange (e) {
    setSymbol (e.target.value);
  }

  function onSubmit (e) {
    e.preventDefault ();
    const t = symbol.trim ().toUpperCase ();
    onAddStock (t);
    setSymbol ('');
  }

  return (
    <form onSubmit={onSubmit}>
      <label htmlFor='symbolText'>Symbol:</label>
      {' '}
      <input
        type='text'
        id='symbolText'
        value={symbol}
        style={{ width: '60px' }}
        onChange={onChange}
      />
      {' '}
      <button
        type='submit'
        disabled={(symbol.trim () === '')}
      >
        Track
      </button>
    </form>
  );
};

StockEntry.propTypes = {
  onAddStock: PropTypes.func.isRequired,
};
