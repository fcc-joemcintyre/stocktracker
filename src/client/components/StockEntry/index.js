import { StockEntry } from './StockEntry';

export { StockEntry };
