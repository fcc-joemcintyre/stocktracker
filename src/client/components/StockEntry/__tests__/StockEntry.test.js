/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { StockEntry } from '../StockEntry';

test ('StockEntry initial display', () => {
  const component = renderer.create (
    <StockEntry onAddStock={() => { /* */ }} />
  );
  const tree = component.toJSON ();
  expect (tree).toMatchSnapshot ();
});

test ('Add stock event', () => {
  const onAddStock = jest.fn ();
  const component = mount (<StockEntry onAddStock={onAddStock} />);
  expect (component.props ().onAddStock).toBeDefined ();

  const input = component.find ('input').first ();
  const button = component.find ('button').first ();
  expect (input).toBeDefined ();
  expect (button).toBeDefined ();

  input.simulate ('change', { target: { value: 'GE' } });
  button.simulate ('submit');
  expect (onAddStock).toBeCalledWith ('GE');
});
