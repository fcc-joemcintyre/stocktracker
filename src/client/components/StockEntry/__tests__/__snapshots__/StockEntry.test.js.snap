// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StockEntry initial display 1`] = `
<form
  onSubmit={[Function]}
>
  <label
    htmlFor="symbolText"
  >
    Symbol:
  </label>
   
  <input
    id="symbolText"
    onChange={[Function]}
    style={
      Object {
        "width": "60px",
      }
    }
    type="text"
    value=""
  />
   
  <button
    disabled={true}
    type="submit"
  >
    Track
  </button>
</form>
`;
